external-crates/move/crates/move-binary-format/src/file_format_common.rs

```diff
@@ -19,6 +19,54 @@ use std::{
     mem::size_of,
 };
 
+// Static assertions about the encoding of the flavor into the version of the binary format.
+const _: () = {
+    let x = BinaryFlavor::shift_and_flavor(0u32);
+    assert!(x == 0x0500_0000u32);
+    assert!(BinaryFlavor::mask_and_shift_to_unflavor(x) == BinaryFlavor::SUI_FLAVOR);
+};
+
+/// Encoding of a the flavor into the version of the binary format for versions >= 7.
+pub struct BinaryFlavor;
+impl BinaryFlavor {
+    pub const FLAVOR_MASK: u32 = 0xFF00_0000;
+    pub const VERSION_MASK: u32 = 0x00FF_FFFF;
+    // The Sui flavor is 0x05
+    pub const SUI_FLAVOR: u8 = 0x05;
+    const SHIFT_AMOUNT: u8 = 24;
```

Comment:
I wonder if there is a more-canonical way to do this in Rust, since we're doing it a few places now.